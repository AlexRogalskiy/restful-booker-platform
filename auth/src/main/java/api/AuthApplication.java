package api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan()
public class AuthApplication {

    // Testing the lookup by making a change in search
    public static void main(String[] args) {
        SpringApplication.run(AuthApplication.class, args);
    }

}
