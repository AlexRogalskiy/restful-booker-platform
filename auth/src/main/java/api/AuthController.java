package api;

import auth.Tokens;
import model.Auth;
import model.Token;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@RestController
public class AuthController {

    @Bean
    public WebMvcConfigurer corsConfigurer() {
        return new WebMvcConfigurer() {
            @Override
            public void addCorsMappings(CorsRegistry registry) {
                String originHost = "http://localhost:3003";

                if(System.getenv("cors") != null){
                    originHost = System.getenv("cors");
                }

                registry.addMapping("/*")
                        .allowedOrigins(originHost)
                        .allowCredentials(true);
            }
        };
    }

    @RequestMapping(value = "/login", method = RequestMethod.POST)
    public ResponseEntity<Token> createToken(@RequestBody Auth auth) {
        if(auth.getUsername().equals("admin") && auth.getPassword().equals("password")){
            return ResponseEntity.ok(new Token(Tokens.create()));
        } else {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).body(null);
        }
    }

    @RequestMapping(value = "/validate", method = RequestMethod.POST)
    public ResponseEntity<Token> validateToken(@RequestBody Token token) {
        if(Tokens.verify(token.getToken())){
            return ResponseEntity.status(HttpStatus.OK).build();
        } else {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
        }
    }

    @RequestMapping(value = "/logout", method = RequestMethod.POST)
    public ResponseEntity clearToken(@RequestBody Token token) {
        Tokens.clear(token.getToken());

        return ResponseEntity.status(HttpStatus.OK).build();
    }

}
