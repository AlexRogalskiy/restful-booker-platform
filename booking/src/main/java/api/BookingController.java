package api;

import db.BookingDB;
import model.Booking;
import model.BookingResults;
import model.CreatedBooking;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import requests.AuthRequests;

import java.sql.SQLException;
import java.util.Optional;

@RestController
public class BookingController {

    BookingDB bookingDB;
    AuthRequests authRequests;

    public BookingController() throws SQLException {
        bookingDB = new BookingDB();
        authRequests = new AuthRequests();
    }

    @CrossOrigin
    @RequestMapping(value = "/", method = RequestMethod.GET)
    public ResponseEntity<BookingResults> getBookings(@RequestParam("roomid") Optional<String> roomid, @RequestParam("keyword") Optional<String> keyword) throws SQLException {
        if(roomid.isPresent()){
            BookingResults searchResults = new BookingResults(bookingDB.queryBookingsById(roomid.get()));
            return ResponseEntity.ok(searchResults);
        }

        if(keyword.isPresent()){
            BookingResults searchResults = new BookingResults(bookingDB.queryBookingsByName(keyword.get()));
            return ResponseEntity.ok(searchResults);
        }

        return ResponseEntity.ok().build();
    }

    @CrossOrigin
    @RequestMapping(value = "/", method = RequestMethod.POST)
    public ResponseEntity<CreatedBooking> createBooking(@RequestBody Booking booking, @CookieValue(value ="token", required = false) String token) throws SQLException {
        if(authRequests.postCheckAuth(token)){
            CreatedBooking body = bookingDB.create(booking);
            return ResponseEntity.ok(body);
        } else {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
        }
    }

    @CrossOrigin
    @RequestMapping(value = "/{id}", method = RequestMethod.GET)
    public Booking getBooking(@PathVariable(value = "id") int id) throws SQLException {
        return bookingDB.query(id);
    }

    @CrossOrigin
    @RequestMapping(value = "/{id}", method = RequestMethod.DELETE)
    public ResponseEntity deleteBooking(@PathVariable(value = "id") int id, @CookieValue(value ="token", required = false) String token) throws SQLException {
        if(authRequests.postCheckAuth(token)){
            if(bookingDB.delete(id)){
                return ResponseEntity.accepted().build();
            } else {
                return ResponseEntity.notFound().build();
            }
        } else {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
        }
    }

    @CrossOrigin
    @RequestMapping(value = "/{id}", method = RequestMethod.PUT)
    public ResponseEntity<CreatedBooking> updateBooking(@RequestBody Booking booking, @PathVariable(value = "id") int id, @CookieValue(value ="token", required = false) String token) throws SQLException {
        if(authRequests.postCheckAuth(token)){
            return ResponseEntity.ok(bookingDB.update(id, booking));
        } else {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
        }
    }

}
