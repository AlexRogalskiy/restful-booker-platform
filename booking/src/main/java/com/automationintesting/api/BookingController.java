package com.automationintesting.api;

import com.automationintesting.db.BookingDB;
import com.automationintesting.model.Booking;
import com.automationintesting.model.BookingResults;
import com.automationintesting.model.CreatedBooking;
import com.automationintesting.model.Message;
import com.automationintesting.requests.MessageRequests;
import com.automationintesting.utils.MessageBuilder;
import org.springframework.context.annotation.Bean;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
import com.automationintesting.requests.AuthRequests;
import com.automationintesting.utils.DatabaseScheduler;
import com.automationintesting.validators.DateCheckValidator;

import javax.validation.Valid;
import java.sql.SQLException;
import java.util.Optional;
import java.util.concurrent.TimeUnit;

@RestController
public class BookingController {

    private BookingDB bookingDB;
    private AuthRequests authRequests;
    private MessageRequests messageRequests;
    private DateCheckValidator dateCheckValidator;

    @Bean
    public WebMvcConfigurer configurer() {
        DatabaseScheduler databaseScheduler = new DatabaseScheduler();
        databaseScheduler.startScheduler(bookingDB, TimeUnit.MINUTES);

        return new WebMvcConfigurer() {
            @Override
            public void addCorsMappings(CorsRegistry registry) {
                String originHost = "http://localhost:3003";

                if(System.getenv("cors") != null){
                    originHost = System.getenv("cors");
                }

                registry.addMapping("/*")
                        .allowedMethods("GET", "POST", "DELETE", "PUT")
                        .allowedOrigins(originHost)
                        .allowCredentials(true);
            }
        };
    }

    public BookingController() throws SQLException {
        bookingDB = new BookingDB(true);
        authRequests = new AuthRequests();
        messageRequests = new MessageRequests();
        dateCheckValidator = new DateCheckValidator();
    }

    @RequestMapping(value = "/", method = RequestMethod.GET)
    public ResponseEntity<BookingResults> getBookings(@RequestParam("roomid") Optional<String> roomid) throws SQLException {
        if(roomid.isPresent()){
            BookingResults searchResults = new BookingResults(bookingDB.queryBookingsById(roomid.get()));
            return ResponseEntity.ok(searchResults);
        }

        return ResponseEntity.ok().build();
    }

    @RequestMapping(value = "/", method = RequestMethod.POST)
    public ResponseEntity<?> createBooking(@Valid @RequestBody Booking booking, @CookieValue(value ="token", required = false) String token) throws SQLException {
        if(dateCheckValidator.isValid(booking.getBookingDates())) {
            if (bookingDB.checkForBookingConflict(booking)) {
                return ResponseEntity.status(HttpStatus.CONFLICT).build();
            } else {
                CreatedBooking body = bookingDB.create(booking);

                MessageBuilder messageBuilder = new MessageBuilder();
                Message message = messageBuilder.build(booking);
                messageRequests.postMessage(message);

                return ResponseEntity.ok(body);
            }
        } else {
            return ResponseEntity.status(HttpStatus.CONFLICT).build();
        }
    }

    @RequestMapping(value = "/{id:[0-9]*}", method = RequestMethod.GET)
    public Booking getBooking(@PathVariable(value = "id") int id) throws SQLException {
        return bookingDB.query(id);
    }

    @RequestMapping(value = "/{id:[0-9]*}", method = RequestMethod.DELETE)
    public ResponseEntity deleteBooking(@PathVariable(value = "id") int id, @CookieValue(value ="token", required = false) String token) throws SQLException {
        if(authRequests.postCheckAuth(token)){
            if(bookingDB.delete(id)){
                return ResponseEntity.accepted().build();
            } else {
                return ResponseEntity.notFound().build();
            }
        } else {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
        }
    }

    @RequestMapping(value = "/{id:[0-9]*}", method = RequestMethod.PUT)
    public ResponseEntity<CreatedBooking> updateBooking(@Valid @RequestBody Booking booking, @PathVariable(value = "id") int id, @CookieValue(value ="token", required = false) String token) throws SQLException {
        if(authRequests.postCheckAuth(token)){
            return ResponseEntity.ok(bookingDB.update(id, booking));
        } else {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
        }
    }

}
