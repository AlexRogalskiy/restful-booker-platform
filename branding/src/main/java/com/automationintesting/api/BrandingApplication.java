package com.automationintesting.api;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan()
public class BrandingApplication {

    public static void main(String[] args) {
        SpringApplication.run(BrandingApplication.class, args);
    }

}
