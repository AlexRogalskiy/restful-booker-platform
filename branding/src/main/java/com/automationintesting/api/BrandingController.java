package com.automationintesting.api;

import com.automationintesting.db.BrandingDB;
import com.automationintesting.model.Branding;
import com.automationintesting.requests.AuthRequests;
import org.springframework.context.annotation.Bean;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import java.sql.SQLException;

@RestController
public class BrandingController {

    private BrandingDB brandingDB;
    private AuthRequests authRequest;

    @Bean
    public WebMvcConfigurer configurer() {

        return new WebMvcConfigurer() {
            @Override
            public void addCorsMappings(CorsRegistry registry) {
                String originHost = "http://localhost:3003";

                if(System.getenv("cors") != null){
                    originHost = System.getenv("cors");
                }

                registry.addMapping("/*")
                        .allowedMethods("GET", "POST", "DELETE", "PUT")
                        .allowedOrigins(originHost)
                        .allowCredentials(true);
            }
        };
    }

    public BrandingController() throws SQLException {
        this.brandingDB = new BrandingDB();
        this.authRequest = new AuthRequests();
    }

    @RequestMapping(value = "/", method = RequestMethod.GET)
    public ResponseEntity<Branding> getBranding() throws SQLException {
        Branding branding = brandingDB.queryBranding();

        return ResponseEntity.ok(branding);
    }

    @RequestMapping(value = "/", method = RequestMethod.PUT)
    public ResponseEntity<?> createBooking(@RequestBody Branding branding, @CookieValue(value ="token", required = false) String token) throws SQLException {
        if(authRequest.postCheckAuth(token)){
            Branding updatedBranding = brandingDB.update(branding);

            return ResponseEntity.ok(updatedBranding);
        } else {
            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
        }
    }

}
