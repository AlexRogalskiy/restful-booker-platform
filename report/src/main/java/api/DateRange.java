package api;

import model.report.RoomReportDate;

import java.util.Calendar;
import java.util.Date;
import java.util.GregorianCalendar;
import java.util.List;

public class DateRange {

    public static List<RoomReportDate> parse(List<RoomReportDate> dates, Date startdate, Date enddate) {
        Calendar calendar = new GregorianCalendar();
        calendar.setTime(startdate);

        while (calendar.getTime().before(enddate)) {
            Date result = calendar.getTime();
            dates.add(new RoomReportDate(result));
            calendar.add(Calendar.DATE, 1);
        }

        dates.add(new RoomReportDate(enddate));

        return dates;
    }

}
