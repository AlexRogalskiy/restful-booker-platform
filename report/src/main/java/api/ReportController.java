package api;

import model.report.Entry;
import model.report.Report;
import model.room.Booking;
import model.room.Bookings;
import model.room.Room;
import org.springframework.context.annotation.Bean;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
import requests.BookingRequests;
import requests.RoomRequests;


import java.sql.SQLException;
import java.util.*;

@RestController
public class ReportController {

    private RoomRequests roomRequests;
    private BookingRequests bookingRequests;

    @Bean
    public WebMvcConfigurer corsConfigurer() {
        return new WebMvcConfigurer() {
            @Override
            public void addCorsMappings(CorsRegistry registry) {
                String originHost = "http://localhost:3003";

                if(System.getenv("cors") != null){
                    originHost = System.getenv("cors");
                }

                registry.addMapping("/*")
                        .allowedOrigins(originHost)
                        .allowCredentials(true);
            }
        };
    }

    public ReportController() throws SQLException {
        roomRequests = new RoomRequests();
        bookingRequests = new BookingRequests();
    }

    @RequestMapping(value = "/", method = RequestMethod.GET)
    public Report getReport() {
        List<Room> rooms = roomRequests.searchForRooms().getBody().getRooms();
        List<Entry> parsedRooms = new ArrayList<>();

        for(Room r : rooms){
            Bookings roomBookings = bookingRequests.getBookings(r.getRoomid()).getBody();

            for(Booking b : roomBookings.getBookings()){
                Entry entry = new Entry(b.getBookingDates().getCheckin(), b.getBookingDates().getCheckout(), b.getFirstname() + " " + b.getLastname() + " - Room: " + r.getRoomNumber());
                parsedRooms.add(entry);
            }
        }

        return new Report(parsedRooms);
    }

}
