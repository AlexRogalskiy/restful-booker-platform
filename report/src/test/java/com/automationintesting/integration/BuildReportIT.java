package com.automationintesting.integration;

import com.automationintesting.api.ReportApplication;
import com.github.tomakehurst.wiremock.common.ConsoleNotifier;
import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
import com.github.tomakehurst.wiremock.junit.WireMockRule;
import io.restassured.response.Response;
import org.approvaltests.Approvals;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
import static com.github.tomakehurst.wiremock.client.WireMock.get;
import static io.restassured.RestAssured.given;

@RunWith(SpringRunner.class)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT, classes = ReportApplication.class)
@ActiveProfiles("dev")
public class BuildReportIT {

    @Rule
    public WireMockRule roomApi = new WireMockRule(WireMockConfiguration.options().notifier(new ConsoleNotifier(true)).port(3001));

    @Rule
    public WireMockRule bookingApi = new WireMockRule(3000);

    @Before
    public void setupStubs(){
        roomApi.stubFor(get("/room")
                .willReturn(
                    aResponse()
                        .withHeader("Content-Type","application/json")
                        .withBody("{\"rooms\":[{\"roomid\":1,\"roomNumber\":101,\"type\":\"Twin\",\"beds\":2,\"accessible\":false,\"details\":\"Wifi, TV, Mini-bar\"}, {\"roomid\":2,\"roomNumber\":102,\"type\":\"Single\",\"beds\":3,\"accessible\":true,\"details\":\"Wifi, TV, Mini-bar\"}]}")
                        .withStatus(200)));

        bookingApi.stubFor(get("/booking/?roomid=1")
                .willReturn(
                    aResponse()
                        .withHeader("Content-Type", "application/json")
                        .withBody("{\"bookings\":[{\"bookingid\":1,\"roomid\":1,\"firstname\":\"James\",\"lastname\":\"Dean\",\"totalprice\":100,\"depositpaid\":true,\"bookingdates\":{\"checkin\":\"2018-01-01\",\"checkout\":\"2018-01-05\"}},{\"bookingid\":2,\"roomid\":1,\"firstname\":\"Mark\",\"lastname\":\"Winteringham\",\"totalprice\":200,\"depositpaid\":false,\"bookingdates\":{\"checkin\":\"2018-02-01\",\"checkout\":\"2018-02-05\"}}]}")
                        .withStatus(200)));

        bookingApi.stubFor(get("/booking/?roomid=2")
                .willReturn(
                    aResponse()
                            .withHeader("Content-Type", "application/json")
                            .withBody("{\"bookings\":[{\"bookingid\":1,\"roomid\":2,\"firstname\":\"James\",\"lastname\":\"Dean\",\"totalprice\":100,\"depositpaid\":true,\"bookingdates\":{\"checkin\":\"2018-03-01\",\"checkout\":\"2018-03-05\"}},{\"bookingid\":2,\"roomid\":2,\"firstname\":\"Mark\",\"lastname\":\"Winteringham\",\"totalprice\":200,\"depositpaid\":false,\"bookingdates\":{\"checkin\":\"2018-04-01\",\"checkout\":\"2018-04-05\"}}]}")
                            .withStatus(200)));

    }

    @Test
    public void testReportCreation(){
        Response reportResponse = given()
                                    .get("http://localhost:3005/report");

        Approvals.verify(reportResponse.body().prettyPrint());
    }

    @Test
    public void testSpecificRoomReportCreation(){
        Response reportResponse = given()
                                    .get("http://localhost:3005/report/room/1");

        Approvals.verify(reportResponse.getBody().prettyPrint());
    }

}
