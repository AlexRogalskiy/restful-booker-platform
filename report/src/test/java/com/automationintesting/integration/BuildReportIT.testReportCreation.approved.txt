{
    "report": [
        {
            "room": "101",
            "values": [
                {
                    "date": "2018-01-01"
                },
                {
                    "date": "2018-01-02"
                },
                {
                    "date": "2018-01-03"
                },
                {
                    "date": "2018-01-04"
                },
                {
                    "date": "2018-01-05"
                },
                {
                    "date": "2018-02-01"
                },
                {
                    "date": "2018-02-02"
                },
                {
                    "date": "2018-02-03"
                },
                {
                    "date": "2018-02-04"
                },
                {
                    "date": "2018-02-05"
                }
            ]
        },
        {
            "room": "102",
            "values": [
                {
                    "date": "2018-03-01"
                },
                {
                    "date": "2018-03-02"
                },
                {
                    "date": "2018-03-03"
                },
                {
                    "date": "2018-03-04"
                },
                {
                    "date": "2018-03-05"
                },
                {
                    "date": "2018-04-01"
                },
                {
                    "date": "2018-04-02"
                },
                {
                    "date": "2018-04-03"
                },
                {
                    "date": "2018-04-04"
                },
                {
                    "date": "2018-04-05"
                }
            ]
        }
    ]
}